pkgs/os-specific/linux/realmd/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib, stdenv
+, autoreconfHook
+, docbook_xml_dtd_412
+, docbook_xml_dtd_42
+, docbook_xml_dtd_43
+, docbook_xsl
+, fetchFromGitLab
+, glib
+, intltool
+, libkrb5
+, libtool
+, libxslt
+, openldap
+, pkg-config
+, polkit
+, systemd
+, xmlto
+}:
+# doc-related (--disable-doc): docbook_xml_dtd_412 docbook_xml_dtd_42 docbook_xml_dtd_43 docbook_xsl libxslt xmlto
+
+stdenv.mkDerivation rec {
+  pname = "realmd";
+  version = "0.17.0";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "1c6q2a86kk2f1akzc36nh52hfwsmmc0mbp6ayyjxj4zsyk9zx5bf";
+  };
+
+  configureFlags = [
+    "--with-distro=NixOS"
+    "--with-systemd-unit-dir=${placeholder "out"}/lib/systemd/system"
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    docbook_xml_dtd_412
+    docbook_xml_dtd_42
+    docbook_xml_dtd_43
+    docbook_xsl
+    glib
+    intltool
+    libkrb5
+    libtool
+    libxslt
+    openldap
+    pkg-config
+    polkit
+    systemd
+    xmlto
+  ];
+
+  preConfigure = ''
+    # service/Makefile.am expects to find the file "service/realmd-NixOS.conf", which:
+    # - isn't provided by upstream
+    # - doesn't make sense in this context and should rather be provided by a module configuring the realmd service
+    sed -i /DISTRO/d service/Makefile.am
+    # don't install "/var/cache" and "/var/lib" - this is provided by the service unit at runtime
+    sed -i '/INSTALL_PROGRAM.\+cachedir/d' service/Makefile.am
+    sed -i '/INSTALL_PROGRAM.\+localstatedir/d' service/Makefile.am
```

Comment:
Can we use a patch or substituteInPlace?

Reply:
Using a patch now, `substituteInPlace` looked messy in this case…